sklearn/ensemble/forest.py

```diff
@@ -72,17 +73,35 @@ class calls the ``fit`` method of each sub-estimator on random samples
 MAX_INT = np.iinfo(np.int32).max
 
 
-def _generate_sample_indices(random_state, n_samples):
+def _generate_sample_indices(random_state, n_samples, max_samples=None):
     """Private function used to _parallel_build_trees function."""
+
+    # Validate `max_samples`
+    if max_samples is None:
+        max_samples = n_samples
+    elif isinstance(max_samples, numbers.Integral):
+        if not (0 < max_samples <= n_samples):
+            msg = "`max_samples` must be in range 1 ... {} but got value {}"
+            raise ValueError(msg.format(n_samples, max_samples))
+    elif isinstance(max_samples, numbers.Real):
+        if not (0 < max_samples <= 1.0):
```

Comment:
I think we should refuse to support max_samples=1 rather than interpret the ambiguous value based on type

Reply:
Good point, since 1.0 only provides a *third* way to indicate to use the entire sample set. I made this change in https://github.com/scikit-learn/scikit-learn/pull/14682/commits/2507d5b44f278047eae2cb1c0bea4705bb266622